### ISSUE TITLE
Masking with tensor input for variable length input/output

### ISSUE THREAD

**Issue by joelkuiper** [fchollet/keras#7511]
I'm trying to build a sequence to sequence classifier for natural language tagging. One common problem is the variable length sequences of sentences (both in input and in output). Now I'm trying to use the Masking feature, but I'm a bit unsure how this works without and Embedding Layer. Basically I feed the input the tensor of (samples, timestep, embedding) directly, so the input is not of (1, vocab_dim) but rather (1, embedding_dim) (in this case 128). How does this work with masking, does it work if I set all the dimensions to zero, or is there something else I need to do?

**Comment by mahnerak** [comment]
First, as always, in this case also you have to feed your model with numpy arrays.
Consider you have sentences of different lengths (List of numpy arrays of different lengths, containing word vocab ids). You have to convert the list to numpy array by padding the arrays.
Look at https://keras.io/preprocessing/sequence/#pad_sequences
```python
keras.preprocessing.sequence.pad_sequences(sequences, dtype='int32')
```
By default, it fills empty parts with 0 value.

Then, you need to modify the network. You have to define Input, then define the Embedding layer with masking (When using Embedding layer you have to use argument mask_zeros instead of adding Masking layer after that).
Example of network:
```python
sequences = Input(shape=(None,), dtype='int32') 
embeddings = Embedding(input_dim=127, output_dim=50, mask_zero=True) (sequences)
hidden_layer = Bidirectional(GRU(units=50)) (embeddings)
predictions = Dense(units=10, activation='softmax') (hidden_layer)

model = Model(inputs=sequences, outputs=predictions)
```
